src/network/network_gaming_messages.cc

```diff
@@ -39,126 +91,55 @@ const std::string NetworkGamingMessages::get_message(const std::string& code,
                                                      const std::string& arg1,
                                                      const std::string& arg2,
                                                      const std::string& arg3) {
+
 	if (ngmessages.find(code) == ngmessages.end()) {
 		return (
 		   format("%s, %s, %s, %s", code, get_message(arg1), get_message(arg2), get_message(arg3)));
 	}
 
-	// push code and all arguments - if existing - in a vector for easier handling
-	std::vector<std::string> strings;
-	strings.push_back(get_message(code));
-	if (!arg1.empty()) {
-		strings.push_back(get_message(arg1));
-		if (!arg2.empty()) {
-			strings.push_back(get_message(arg2));
-			if (!arg3.empty()) {
-				strings.push_back(get_message(arg3));
-			}
-		}
+	// clang-tidy says this shouldn't be const because it may be returned directly in case 0 below
+	std::string msg_translated = _(ngmessages.at(code));
+
+	const size_t n_fmt_arg = format_arguments_count(msg_translated);
+	assert(n_fmt_arg <= 3);
+
+	////// TODO(tothxa): remove checks after testing? (before even merging the PR?)
```

Comment:
This check really shouldn't be needed, we run extensive translation checks before releases to ensure any printf placeholder mismatches in translations are removed. I can remember only one case where one slipped though. And if we do want to check for problems, we should really take a much more general approach since there are lots of translatable strings with placeholders – these here aren't special.

Reply:
I just added the new check because I wanted to somehow include the offending format string for the other 2 checks (missing or extra arguments), and I didn't like any option I was able to come up with. I disliked this one the least, though it doesn't do that, but it makes sure that the problem is not in the translation, so the format can be used from the table in this file.

However the comment also refers to the other 2 checks. They shouldn't be needed either, I just added them in the hope that they might catch some bugs in the call sites that were masked before by the try-catch trickery.

Reply:
Can be merged when the debugging checks are removed